test/deps/executables_test.dart

```diff
@@ -190,56 +190,4 @@ main() {
     test("are listed if --dev flag is set", _testAllDepsOutput('foo:bar'));
     test("are skipped if --no-dev flag is set", _testNonDevDepsOutput('\n'));
   });
-
-  group("lists dependencies overrides", () {
```

Comment:
One thing that would probably be worth adding is a test where you have two versions of `foo` in different directories with different files under bin. Make the original dep be a path dep to one of them, and override it to the other one. You should get a list of the binaries in the overridden one not the original (pretty sure this would pass today).